### ISSUE TITLE
Designator initialization macros not supporting variables

### ISSUE THREAD

**Issue by fairlight1337** [cram2/cram_core#3]
The initialization macros for designators don't support dynamic content anymore.

Formerly, code like this
```lisp
> (a location `((pose ,p)))
```
would result in a location designator with the contents of variable `p` as the value for the key `pose`. With the introduced changes (although I understand that they are meant to address code readability issues) this results in
```lisp
#<location-designator ((:pose :p)) {...}>
```
The functionality got lost due to the fact that the back-quote in the property definition is now unnecessary and all remaining lists are parsed as single pairs. For reference, see [here](https://github.com/cram2/cram_core/blob/master/cram_designators/src/cram-designators/initialization-macros.lisp#L49).

This makes these macros pretty much useless for actual code containing variable values. It is nice for presenting code such that it looks like in certain presentations, but at the same time the functionality went overboard.

Can we change that back to the way they were implemented originally, or at least get back the syntax we had before? I don't mind having keywords instead of symbols for keys, but the variables are a must. The desired use-case looks like this:
```lisp
> (let ((p (tf:make-identity-pose)))
    (a location `((:pose ,p))))
```
Alternatively, we can drop the outer parenthesizes and try something like
```lisp
> (let ((p (tf:make-identity-pose)))
    (a location `(:pose ,p) (:color :red)))
```

Might affect @gaya- and @airballking.

**Comment by airballking** [comment]
To clarify: You report a bug in one or several macros, and ask whether somebody could fix it?

One example of a broken macro being: https://github.com/cram2/cram_core/blob/master/cram_designators/src/cram-designators/initialization-macros.lisp#L49

**Comment by fairlight1337** [comment]
I'm reporting this because it changed while I was absent. I see how the new behavior might benefit certain use-cases while completely changing the effect and actual use of the macro. I'm asking whether someone has thoughts on how this should be handled, as apparently you guys were working on CRAM while I was away. I was referencing the `a` macro; and that's exactly the one you posted the link to.

This stated, can you please clarify your point, @airballking? Or was that the clarification you needed?

**Comment by airballking** [comment]
I was just asking about the nature of this issue because there was no question in your original post. So, you're reporting a bug. I understand that now.

I personally was neither working on the cram0.2 refactoring nor with it. That was @gaya-. So, we maybe she can answer these questions:
  (1) Is this a bug or a feature?
  (2) Is there a nice way to also cover your use case?

**Comment by gaya-** [comment]
There were multiple shortcomings in the old implementation. I will have to take some time to look into this in more detail but the major problem that I had was the inability to support recursive descriptions, such as
`(a location (of (an object (type cup))))`.
The current implementation is my effort to support recursive descriptions.
I did realize at some point that my approach makes supporting variables very difficult and ugly, so I didn't put any more effort into implementing that.
It would definitely be really nice to have that.
I'm not sure how urgent this is.
I guess we will have to discuss that further in more detail before I can commit to any actions.

**Comment by gaya-** [comment]
Addressed in https://github.com/cram2/cram_core/pull/5

**Comment by gaya-** [comment]
Solved in #5
